crates/model/src/bytes_hex.rs

```diff
@@ -28,6 +29,29 @@ where
     hex::decode(hex_str).map_err(D::Error::custom)
 }
 
+pub struct BytesHex(());
```

Comment:
```rust
    #[serde(with = "crate::bytes_hex")]
    pub optimized_call_data: Option<Vec<u8>>,
```
didn't work because `bytes_hex` required `T: AsRef<[u8]>` which `Option<Vec<u8>>` doesn't implement.
Adding `BytesHex` and implementing `(De)SerializeAs` did the trick. The same pattern was already used for `U256` [here](https://github.com/cowprotocol/services/blob/main/crates/model/src/u256_decimal.rs#L6-L24) and can be used more flexibly with container types like this:
```rust
#[serde_as(as = "Vec<BytesHex>")]
#[serde_as(as = "HashMap<_, BytesHex>")]
#[serde_as(as = "Option<BytesHex>")]
```
Thanks for the tip @nlordell.